specs/testing/SkillsMatrixTesting.md

```diff
@@ -0,0 +1,58 @@
+# Automated Testing of Skills
+Defines the short term goals around the automated testing of skills. 
+
+# Goal
+Automate the testing matrix for Bot/Skill interactions. The full testing matrix runs every night. Failures result in emails. 
+
+# Version of the Botbuilder SDK
+1. For each test run, the latest *preview* Bot Builder SDK should be consumed from the MyGet.
+
+2. Alternatly, a specific version of the BotBuilder SDK could be specified via an Azure Devops variable. 
+
+# Testing Matrix
+The following testing matrix needs to be fully tested
+|   |        |    | **Skill**    |        |   |
+|---|--------|----|------------|--------|---|
+|   |        | C# | Javascript | Python |   |
+|   | C#     | 1  |     2      | 3      |   |
+|   | JS     | 4  |5           | 6      |   |
+|   | Python | 7  |8           | 9      |   |
+
+1. C# Bot acting as consumer, calling a C# Skill
+2. C# Bot acting as consumer, calling a JS Skill
+3. C# Bot acting as consumer, calling a Pyton Skill
+4. JS Bot acting as consumer, calling a C# Skill
+5. JS Bot acting as consumer, calling a JS Skill
+6. JS Bot acting as consumer, calling a Python Skill
+7. Python Bot acting as consumer, calling a C# Skill
+8. Python Bot acting as consumer, calling a JS Skill
+9. Python Bot acting as consumer, calling a Python Skill
+
+# Tests
+V0 of the tests is an Echo. The Consumer bot will send a Message containing a GUID to the skill, and the skill encho that back. 
+
+# Bots we need
```

Comment:
I wonder if testing OAuth card / SSO would be a valid P1. Maybe we could even get help from the service team on the implementation? perhaps?

Reply:
This is a feature that can only be tested in integration tests, and the unit test coverage is lackluster. Seems like a great opportunity to gain confidence on this codepath.

Reply:
Let's sync on OAuth. We need the testing, and I don't know if it belongs here or in it's own project.